arch/x86/include/kernel_arch_data.h

```diff
@@ -39,6 +39,13 @@
 #include <misc/dlist.h>
 #endif
 
+/* Some configurations require that the stack/registers be adjusted before
+ * _thread_entry. See discussion in swap.S for _x86_thread_entry_wrapper()
+ */
+#if defined(CONFIG_X86_IAMCU) || defined(CONFIG_DEBUG_INFO)
+#define _WRAPPER_REQUIRED
```

Comment:
Since this is outside thread.c, I'd change `_WRAPPER_REQUIRED` to `_THREAD_WRAPPER_REQUIRED` or something along those lines.